File: LeetCode/1_TWO_SUM.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {nums[i]: i for i in range(0, len(nums))}
        for i in range(len(nums)):
            if hash.get(target - nums[i]) is not None and hash.get(target - nums[i]) != i:
                return [i, hash.get(target - nums[i])]
        return [0, 1]


arr = [0, 4, 3, 0]
solution = Solution();
print(solution.twoSum(arr, 0))

File: LeetCode/1_Two_Sum/1_Two_Sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val cache: MutableMap<Int, Int> = mutableMapOf()
        for ((i, one) in nums.withIndex()) {
            cache.put(one, i)
        }

        for ((i, one) in nums.withIndex()) {
            val two = target - one
            val j: Int? = cache.get(two)
            if (j != null && i != j) {
                return intArrayOf(i, j)
            }
        }
        return intArrayOf(0, 1)
    }
}

fun main() {
    val nums = intArrayOf(3,2,4)
    val target = 6

    val s = Solution()
    val result = s.twoSum(nums, target)
    println("[${result[0]}, ${result[1]}]")
}
